websrc/source/documents/tutorials/contiv-policy-kubernetes-16.md

```diff
@@ -224,29 +233,418 @@ PING 10.1.1.2 (10.1.1.2): 56 data bytes
+Rule  Priority  To EndpointGroup  To Network  To IpAddress  Protocol  Port  Action
+----  --------  ----------------  ----------  ---------     --------  ----  ------
+```
+Now check that from epgB, only TCP 8001 port is open. To test this, let's run `iperf` on bpod and verify using the `nc` utility on apod.
+
+On bpod:
+
+```
+[vagrant@kubeadm-master ~]$ kubectl exec -it bpod sh
+/ # iperf -s -p 8001
+------------------------------------------------------------
+Server listening on TCP port 8001
+TCP window size: 85.3 KByte (default)
+------------------------------------------------------------
+```
+Open another terminal and login to apod:
+
+```
+[vagrant@kubeadm-master ~]$ kubectl exec -it apod sh
+/ # nc -zvw 1 10.1.1.2 8001     # here 10.1.1.2 is IP address of bpod.
+10.1.1.2 (10.1.1.2:8001) open
+/ # nc -zvw 1 10.1.1.2 8000
+nc: 10.1.1.2 (10.1.1.2:8000): Operation timed out
+/ # exit
+```
+We can see that port 8001 is open and port 8000 is not open. Exit bpod as well.
+
+```
+/ # iperf -s -p 8001
+------------------------------------------------------------
+Server listening on TCP port 8001
+TCP window size: 85.3 KByte (default)
+------------------------------------------------------------
+[  4] local 10.1.1.2 port 8001 connected with 10.1.1.1 port 38137
+[ ID] Interval       Transfer     Bandwidth
+[  4]  0.0- 0.0 sec  0.00 Bytes  0.00 bits/sec
+^C/ # exit
+```
+
+### <a name="ch3"></a> Chapter 3 - Bandwidth Policy
+
+In this chapter, we will explore the bandwidth policy feature of Contiv. We will create a tenant, a network and some groups. Then we will attach a netprofile to one endpoint group and verify that the applied bandwidth policy works as expected.
+
+So, let's create a tenant, a network and group "A" under the network.
+
+```
+[vagrant@kubeadm-master ~]$ netctl tenant create BandwidthTenant
+Creating tenant: BandwidthTenant
+[vagrant@kubeadm-master ~]$ netctl network create --tenant BandwidthTenant --subnet=50.1.1.0/24 --gateway=50.1.1.254 -p 1001 -e "vlan" BandwidthTestNet
+Creating network BandwidthTenant:BandwidthTestNet
+[vagrant@kubeadm-master ~]$ netctl group create -t BandwidthTenant BandwidthTestNet epgA
+Creating EndpointGroup BandwidthTenant:epgA
+```
+```
+[vagrant@kubeadm-master ~]$ netctl net ls -a
+Tenant           Network           Nw Type  Encap type  Packet tag  Subnet        Gateway     IPv6Subnet  IPv6Gateway  Cfgd Tag
+------           -------           -------  ----------  ----------  -------       ------      ----------  -----------  ---------
+default          default-net       data     vxlan       0           20.1.1.0/24   20.1.1.1
+default          contiv-net        data     vxlan       0           10.1.2.0/24   10.1.2.1
+blue             contiv-net        data     vxlan       0           10.1.2.0/24   10.1.2.1
+TestTenant       TestNet           data     vlan        0           10.1.1.0/24   10.1.1.254
+BandwidthTenant  BandwidthTestNet  data     vlan        1001        50.1.1.0/24   50.1.1.254
```

Comment:
This tutorial is a continuation of the networking 101 tutorial which is why all the networks are shown. Should I still only show the vlan networks?